ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/LambdaAnalyzer.fs

```diff
@@ -84,8 +86,30 @@ type LambdaAnalyzer() =
             | x -> hasMatches, i = pats.Count, x
 
         compareArgsRec expr 0 null
-
-    let isApplicable (expr: IFSharpExpression) =
+    
+    let isLambdaApplicable (lambda: ILambdaExpr) =
```

Comment:
I don't think this name is right for the check. It doesn't check the lambda in any way, instead it checks the _context_ where the lambda happens to be, and the only context constraint it adds is lambda shouldn't be used as an argument to a member where an impicit conversion would happen. Could we show it in the name instead?